jlm/util/intrusive-list.hpp

```diff
@@ -443,6 +452,9 @@ class IntrusiveList
       return;
     }
 
+    // Maintaining the sizes of the linked lists requires range splicing to be O(n)
+    size_t count = std::distance(begin, end);
```

Comment:
Stupid question: Is slicing anywhere actually used? In other words, do we pay a performance penalty here?